src/rating.tsx

```diff
@@ -0,0 +1,332 @@
+        color: vars.colors.errorHigh,
+    },
+    {
+        ActiveIcon: IconFaceSlightlySadFilled,
+        InactiveIcon: IconFaceSlightlySadRegular,
+        color: vars.colors.error,
+    },
+    {
+        ActiveIcon: IconFaceNeutralFilled,
+        InactiveIcon: IconFaceNeutralRegular,
+        color: vars.colors.warning,
+    },
+    {
+        ActiveIcon: IconFaceHappyFilled,
+        InactiveIcon: IconFaceHappyRegular,
+        color: vars.colors.success,
+    },
+    {
+        ActiveIcon: IconFaceSuperHappyFilled,
+        InactiveIcon: IconFaceSuperHappyRegular,
+        color: vars.colors.successHigh,
+    },
+];
+
+interface BaseRatingProps {
+    size?: number;
+    dataAttributes?: DataAttributes;
+    valueLabels?: Array<string>;
+    'aria-label'?: string;
+    'aria-labelledby'?: string;
+}
+
+interface QuantitativeRatingProps extends BaseRatingProps {
+    type?: 'quantitative';
+    icon?: RatingIconProps;
+    count?: number;
+    icons?: never;
+}
+
+interface QualitativeRatingProps extends BaseRatingProps {
+    type: 'qualitative';
+    icons?: Array<RatingIconProps>;
+    icon?: never;
+    count?: never;
+}
+
+type RatingProps = ExclusifyUnion<QualitativeRatingProps | QuantitativeRatingProps> & {
+    value?: number;
+    defaultValue?: number;
+    onChangeValue?: (value: number) => void;
+    disabled?: boolean;
+};
+
+type InfoRatingProps = Omit<QuantitativeRatingProps, 'type' | 'valueLabels'> & {
+    value?: number;
+};
+
+type InternalRatingProps = ExclusifyUnion<RatingProps | InfoRatingProps> & {
+    isInteractive: boolean;
+    role: 'radiogroup' | 'img';
+};
+
+const useRatingState = ({
+    value,
+    defaultValue,
+    iconsCount,
+    onChangeValue,
+}: {
+    value?: number;
+    defaultValue?: number;
+    iconsCount: number;
+    onChangeValue?: (value: number) => void;
+}): [number, (value: number) => void] => {
+    const isControlledByParent = value !== undefined;
+
+    const getValueInRange = React.useCallback(
+        (value?: number) => {
+            return value === undefined ? 0 : Math.max(0, Math.min(iconsCount, value));
+        },
+        [iconsCount]
+    );
+
+    const [currentValue, setCurrentValue] = React.useState<number>(getValueInRange(defaultValue));
+
+    const updateValue = (newValue: number) => {
+        if (!isControlledByParent) {
+            setCurrentValue(newValue);
+        }
+        onChangeValue?.(newValue);
+    };
+
+    return [isControlledByParent ? getValueInRange(value) : currentValue, updateValue];
+};
+
+const InternalRating: React.FC<InternalRatingProps> = ({
+    icons = DEFAULT_QUALITATIVE_ICONS,
+    count = DEFAULT_RATING_ICON_COUNT,
+    icon = DEFAULT_RATING_ICON,
+    size = DEFAULT_RATING_SIZE,
+    type = 'quantitative',
+    dataAttributes,
+    onChangeValue,
+    defaultValue,
+    value,
+    disabled,
+    role,
+    valueLabels,
+    'aria-label': ariaLabel,
+    'aria-labelledby': ariaLabelledBy,
+}) => {
+    // TODO: get translations
```

Comment:
@aweell we need translations for these default texts (and also for the `x de N` ones). Has this been considered?

Reply:
Discussed offline. We will create translations with Google Translate and create a ticket for loc to review it